src/coreclr/src/tools/Common/JitInterface/JitConfigProvider.cs

```diff
@@ -29,7 +30,12 @@ public static JitConfigProvider Instance
         private Dictionary<string, string> _config = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
         private object _keepAlive; // Keeps callback delegates alive
 
-        public static void Initialize(IEnumerable<CorJitFlag> jitFlags, IEnumerable<KeyValuePair<string, string>> parameters, string jitPath = null)
+        public static void Initialize(
+            TargetOS targetOS,
+            TargetArchitecture targetArchitecture,
+            IEnumerable<CorJitFlag> jitFlags,
+            IEnumerable<KeyValuePair<string, string>> parameters,
+            string jitPath = null)
```

Comment:
Now that you are loading the JIT .dll manually, you should not need the SetDllImportResolver below.

Also, the jitPath won't work anymore with these changes. Should we keep it working?

Reply:
Well, I guess it's probably no big deal to fix CorInfoImpl.Startup to optionally accept the jitPath and use it to override the synthesized subfolder when present. I'll get rid of the SetDllImportResolver based on your suggestion.

Reply:
Cleaned up in 20th commit.

Reply:
The jit path is filename (not directory name) of the .dll to load today. 

The jit path exists for convenience so that you do not have to copy things around if you happen to have JIT .dll someplace. I think it would be best to keep the jit path as a simple filename (no additional substitutions/mangling).

Reply:
Thanks for clarifying, that makes sense to me and I believe this is exactly how I modified the code in 20th commit.

Reply:
👍